backend/services/wasabi_service.py

```
@@ -0,0 +1,667 @@
+                async with httpx.AsyncClient(timeout=120.0) as client:
+                    response = await client.get(url)
+                    response.raise_for_status()
+                    file_bytes = response.content
+                    
+                    # Detect Content-Type from response headers if not provided
+                    if not detected_content_type:
+                        detected_content_type = response.headers.get('Content-Type', 'image/jpeg')
+                        debug_log(f"[S3] Detected Content-Type: {detected_content_type}")
+                    
+                    break
+            except Exception as e:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)